### PR TITLE
Added loading before rendering

### LINKED ISSUE #8597: Loading Animation when a component or page is loading
### Current Behavior
Currently, nothing shows on screen when something loads
image

### Desired Behavior
Implement loading animation when something loads

### Mockups

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by EraKin575**
**Notes for Reviewers**

This PR fixes #8597 

circular progress component renders before chart 
image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by EraKin575** [comment]
@sudhanshutech sorry for the delay. WIll you check if this is ok then I can further apply this to the whole codebase where components take time to render

File: ui/components/Dashboard/ConnectionCharts.js [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 
-
 export default function ConnectionStatsChart({ classes }) {
-
-  const [chartData,setChartData] = useState([])
+  const [chartData, setChartData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);  // New isLoading state
 
   useEffect(() => {
+    setIsLoading(true);  // Set loading state to true when fetch begins
```
Comment (Ckbhatia) [inline ui/components/Dashboard/ConnectionCharts.js:15]: It's not required if you initialize the state with `true`. Also handle an error case as well for better UX

File: ui/components/Dashboard/ConnectionCharts.js [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
     getConnectionStatusSummary().then(json => {
       setChartData(json.connections_status
         .filter(data => isValidColumnName(data.status))
-        .map(data => [data.status,data.count]))
-    })
-  },[])
+        .map(data => [data.status, data.count]));
+      setIsLoading(false);  // Set loading state to false when fetch completes
```
Comment (Ckbhatia) [inline ui/components/Dashboard/ConnectionCharts.js:20]: I don't think, comments are required for this small easily predictable state.

File: ui/components/Dashboard/ConnectionCharts.js [hunk 3]
```diff
@@ -49,36 +50,45 @@ export default function ConnectionStatsChart({ classes }) {
...
-              }}
-            >
-              <Typography style={{ fontSize : "1.5rem", marginBottom : "1rem" }} align="center" >
+      <Box
+        sx={{
+          display : "flex",
+          justifyContent : "center",
+          alignItems : "center",
+          alignContent : "center",
+          height : "300px",  // Fixed height for consistency
```
Comment (Ckbhatia) [inline ui/components/Dashboard/ConnectionCharts.js:66]: No need to add comments for simple things.

**Review by Ckbhatia — CHANGES_REQUESTED** [review]
Please check your prettier for formatting. I see unnecessary changes due to formatting.

File: ui/components/Dashboard/ConnectionCharts.js (hunk 3 above)
Comment (EraKin575) [reply ui/components/Dashboard/ConnectionCharts.js:66]: oof, I forgot to remove them. I had them for my own track of things

File: ui/components/Dashboard/ConnectionCharts.js [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
+    setIsLoading(true);
     getConnectionStatusSummary().then(json => {
       setChartData(json.connections_status
         .filter(data => isValidColumnName(data.status))
-        .map(data => [data.status,data.count]))
-    })
-  },[])
+        .map(data => [data.status, data.count]));
+      setIsLoading(false);
+    });
```
Comment (aabidsofi19) [inline ui/components/Dashboard/ConnectionCharts.js:21]: You are not handling error cases m and also the finally case

File: ui/components/Dashboard/DashboardMeshModelGraph.js [hunk 5]
```diff
@@ -1,153 +1,141 @@
...
   useEffect(() => {
     fetchr().then((json) => {
-      setTotal(json["total_count"])
-    }).catch(e => console.log("Api Error : ", e))
-  }, [])
-  return total
+      setTotal(json["total_count"]);
+      setLoading(false);
+    }).catch(e => {
+      console.log("Api Error : ", e);
```
Comment (aabidsofi19) [inline ui/components/Dashboard/DashboardMeshModelGraph.js:19]: remove logs

Comment (aabidsofi19) [reply ui/components/Dashboard/DashboardMeshModelGraph.js:19]: and add finally case

File: ui/components/Dashboard/DashboardMeshModelGraph.js [hunk 6]
```diff
@@ -1,153 +1,141 @@
...
 
-  const [categoryMap, setCategoryMap] = useState({})
-  const cleanedData = useMemo(() =>
-    Object
-      .keys(categoryMap)
-      .map((key) => ([key, categoryMap[key]]))
-  ,[categoryMap])
-
-  const chartOptions =useMemo(() => ({
+  const cleanedData = useMemo(() => Object.keys(categoryMap).map((key) => ([key, categoryMap[key]])), [categoryMap]);
```
Comment (aabidsofi19) [inline ui/components/Dashboard/DashboardMeshModelGraph.js:74]: formatting has beeen messed up

File: ui/components/Dashboard/DashboardMeshModelGraph.js [hunk 7]
```diff
@@ -1,153 +1,141 @@
...
+      const promises = categoriesJson['categories'].map((category) => {
+        let categoryName = category.name;
+        return getModelFromCategoryApi(categoryName).then((modelsJson) => {
+          setCategoryMap((prevState) => ({ ...prevState, [categoryName] : modelsJson["total_count"] }));
+        });
+      });
+
+      Promise.all(promises).then(() => {
+        setLoading(false);
+      });
```
Comment (aabidsofi19) [inline ui/components/Dashboard/DashboardMeshModelGraph.js:106]: catch , finally

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/Dashboard/DashboardMeshModelGraph.js (hunk 7 above)
Comment (EraKin575) [reply ui/components/Dashboard/DashboardMeshModelGraph.js:106]: @aabidsofi19 shall I use notify hook for handling error cases?

Comment (aabidsofi19) [reply ui/components/Dashboard/DashboardMeshModelGraph.js:106]: yup,make the loading state false in catch and finally

**Review by aabidsofi19 — APPROVED** [review]

**Comment by Ghat0tkach** [comment]
@EraKin575 
Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by sudhanshutech** [comment]
@EraKin575 incorporated to other pages too?

**Comment by sudhanshutech** [comment]
@EraKin575 any updates

**Comment by EraKin575** [comment]
@sudhanshutech  actually Suspense is not supported in this version by server side rendering. I think I have to go to the usual route

**Pull request closed without merging**